File: spec/javascript/test_enum_fancy.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('EnumFancy', 'src/enum_0.bin', function(r, EnumFancy) {
    assert.equal(r.pet1, EnumFancy.Animal.CAT);
    assert.equal(r.pet2, EnumFancy.Animal.CHICKEN);
});

File: spec/java/src/io/kaitai/struct/spec/TestEnumFancy.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.EnumFancy;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestEnumFancy extends CommonSpec {
    @Test
    public void testEnumFancy() throws Exception {
        EnumFancy r = EnumFancy.fromFile(SRC_DIR + "enum_0.bin");

        assertEquals(r.pet1(), EnumFancy.Animal.CAT);
        assertEquals(r.pet2(), EnumFancy.Animal.CHICKEN);
    }
}
